frame/democracy/src/weights.rs

```diff
@@ -199,19 +200,32 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	// Storage: Democracy ReferendumCount (r:1 w:0)
 	// Storage: Democracy ReferendumInfoOf (r:1 w:0)
 	fn on_initialize_base(r: u32, ) -> Weight {
-		(6_259_000 as Weight)
+		(10_159_000 as Weight)
+			// Standard Error: 3_000
+			.saturating_add((5_708_000 as Weight).saturating_mul(r as Weight))
+			.saturating_add(T::DbWeight::get().reads(5 as Weight))
+			.saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(r as Weight)))
+	}
+	// Storage: Democracy LastTabledWasExternal (r:1 w:0)
+	// Storage: Democracy NextExternal (r:1 w:0)
+	// Storage: Democracy PublicProps (r:1 w:0)
+	// Storage: Democracy LowestUnbaked (r:1 w:0)
+	// Storage: Democracy ReferendumCount (r:1 w:0)
+	// Storage: Democracy ReferendumInfoOf (r:1 w:0)
+	fn on_initialize_base_with_launch_period(r: u32, ) -> Weight {
+		(8_550_000 as Weight)
 			// Standard Error: 4_000
-			.saturating_add((5_032_000 as Weight).saturating_mul(r as Weight))
+			.saturating_add((5_753_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(r as Weight)))
```

Comment:
on_initialize_base_with_launch_period do exactly the same number of read and writes as on_initialize_base.
I guess on_initialize_base is also doing the launch period because block number is 0 ?

in order to be more correct we should set the block number to 1 in on_initialize_base.

Reply:
You are right.